stock_storage_type/models/stock_location.py

```diff
@@ -55,6 +54,86 @@ class StockLocation(models.Model):
     )
+    location_is_empty = fields.Boolean(
+        compute="_compute_location_is_empty",
+        store=True,
+        help="technical field: True if the location is empty "
+        "and there is no pending incoming products in the location"
+    )
+
+    in_move_ids = fields.One2many(
+        'stock.move', 'location_dest_id',
+        domain=[
+            ('state', 'in', ('waiting', 'confirmed', 'partially_available', 'assigned'))
+        ],
+        help="technical field: the pending incoming stock.moves in the location",
+    )
+
+    in_move_line_ids = fields.One2many(
+        'stock.move.line', 'location_dest_id',
+        domain=[
+            ('state', 'in', ('waiting', 'confirmed', 'partially_available', 'assigned'))
+        ],
+        help="technical field: the pending incoming "
+        "stock.move.lines in the location",
+    )
+    children_in_move_line_ids = fields.Many2many(
+        'stock.move.line',
+        _compute='_compute_children_in_move_line_ids',
+        store=True,
+        help="technical field: the pending incoming stock.move.lines "
+        "in the location or children locations",
+    )
+    location_will_contain_lot_ids = fields.Many2many(
+        'stock.production.lot',
+        store=True,
+        compute="_compute_location_will_contain_lot_ids",
+        help="technical field: list of stock.production.lots in "
+        "the location, either now or in pending operations"
+    )
+    location_will_contain_product_ids = fields.Many2many(
+        'product.product',
+        store=True,
+        compute="_compute_location_will_contain_product_ids",
+        help="technical field: list of products in "
+        "the location, either now or in pending operations"
+    )
+
+    @api.depends('child_ids.children_in_move_line_ids')
+    def _compute_children_in_move_line_ids(self):
+        for rec in self:
+            rec.children_in_move_line_ids = (
+                rec.mapped('child_ids.children_in_move_line_ids') | rec.in_move_line_ids
+            )
+
+    @api.depends('quant_ids', 'in_move_ids', 'children_in_move_line_ids')
+    def _compute_location_will_contain_product_ids(self):
+        for rec in self:
+            rec.location_will_contain_product_ids = (
+                rec.mapped('quant_ids.product_id')
+                | rec.mapped("in_move_ids.product_id")
+                | rec.mapped("children_in_move_line_ids.product_id")
+            )
+
+    @api.depends('quant_ids', 'children_in_move_line_ids')
+    def _compute_location_will_contain_lot_ids(self):
+        for rec in self:
+            rec.location_will_contain_lot_ids = (
+                rec.mapped('quant_ids.lot_id')
+                | rec.mapped("children_in_move_line_ids.lot_id")
+            )
+
+    @api.depends('quant_ids', 'in_move_ids', 'in_move_line_ids')
+    def _compute_location_is_empty(self):
+        for rec in self:
+            if (
+                    rec.quant_ids
```

Comment:
There could be a quant with quantity = 0. In this case the location is still empty